File: 0031-next-permutation/0031-next-permutation.cpp
class Solution {
public:
    void nextPermutation(vector<int>& nums) {
        bool flag = false;
        for(int i=nums.size()-2; i>=0; i--){
            if(nums[i]<nums[i+1]){
                int number;
                for(int j=nums.size()-1; j>i; j--){
                    if(nums[i] < nums[j]){
                        number = nums[j];
                        nums[j] = nums[i];
                        nums[i] = number;
                        break;
                    }
                }
                sort(nums.begin()+i+1, nums.end());
                flag = true;
                break;
            }
        }
        if(!flag) sort(nums.begin(),nums.end());
    }
};

File: 0031-next-permutation/0031-next-permutation.py
class Solution:
    def nextPermutation(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        flag = False
        for i in (range(n-2,-1,-1)):
            if nums[i] < nums[i+1]:
                for j in (range(n-1,i,-1)):
                    if nums[i] < nums[j]:
                        tmp = nums[i]
                        nums[i] = nums[j]
                        nums[j] = tmp
                        flag = True
                        nums[i+1:] = sorted(nums[i+1:])
                        break
            if flag == True:
                break
        if flag == False:
            nums.sort()